src/tye_lab_to_nwb/ast_neuropixels/convert_session.py

```diff
@@ -0,0 +1,138 @@
+import traceback
+from pathlib import Path
+from typing import Optional, Dict
+from warnings import warn
+
+from nwbinspector import inspect_nwb
+from nwbinspector.inspector_tools import format_messages, save_report
+
+from neuroconv.utils import (
+    FilePathType,
+    FolderPathType,
+    load_dict_from_file,
+    dict_deep_update,
+)
+from tye_lab_to_nwb.ast_neuropixels import AStNeuroPixelsNNWBConverter
+from tye_lab_to_nwb.tools import read_session_config
+
+
+def session_to_nwb(
+    nwbfile_path: FilePathType,
+    ecephys_recording_folder_path: FolderPathType,
+    subject_metadata: Optional[Dict[str, str]] = None,
+    stub_test: Optional[bool] = False,
+):
+    """
+    Converts a single session to NWB.
+
+    Parameters
+    ----------
+    nwbfile_path : FilePathType
+        The file path to the NWB file that will be created.
+    ecephys_recording_folder_path: FolderPathType
+        The path that points to the folder where the Neuropixels files are located.
+    subject_metadata: dict, optional
+        The optional metadata for the experimental subject.
+    stub_test: bool, optional
+        For testing purposes, when stub_test=True only writes a subset of ecephys data.
+        Default is to write the whole ecephys recording to the file.
+    """
+
+    source_data = dict()
+    conversion_options = dict()
+
+    # Add Recording
+    ecephys_recording_folder_path = Path(ecephys_recording_folder_path)
+    spike_glx_ap_bin_file_paths = list(ecephys_recording_folder_path.glob("*.ap.bin"))
```

Comment:
However #23 if they collected more than 1 subject data within the same folder then the folder approach wouldn't help, or based on subject metadata (from the Excel file) I could look up which "`imec{ind}.ap.bin`" to pick from the folder.

Reply:
Changed source data to file path https://github.com/catalystneuro/tye-lab-to-nwb/pull/24/commits/15fbaccc476d167ffe9cf673fadf8b12855b5ce8